libraries/botbuilder-dialogs/src/waterfallDialog.ts

```diff
@@ -246,7 +245,7 @@ export class WaterfallDialog<O extends object = {}> extends Dialog<O> {
                 'InstanceId': instanceId,
             }});
         } else if (reason === DialogReason.cancelCalled) {
-            var index = instance.state[state.stepIndex];
+            var index = state.stepIndex;
```

Comment:
is this correct?

Reply:
@v-kydela can you demonstrate that there is a bug and write a test that shows there is a bug in the code? After we determine there is a bug, we should port the test to the other languages as well.

Reply:
I've rolled back this bug fix so I can put it in a separate PR